flytekit/clis/sdk_in_container/run.py

```diff
@@ -30,115 +31,217 @@
     StructuredDatasetTransformerEngine,
 )
 
+REMOTE_KEY = "remote"
 
-@click.command(
-    context_settings=dict(
-        ignore_unknown_options=True,
-        allow_extra_args=True,
-    ),
-)
-@click.argument(
-    "file_and_workflow",
-)
-@click.option(
-    "--remote",
-    "is_remote",
-    required=False,
-    is_flag=True,
-    default=False,
-)
-@click.option(
-    "-p",
-    "--project",
-    required=False,
-    type=str,
-    default="flytesnacks",
-)
-@click.option(
-    "-d",
-    "--domain",
-    required=False,
-    type=str,
-    default="development",
-)
-@click.option(
-    "--destination-dir",
-    "destination_dir",
-    required=False,
-    type=str,
-    default="/root",
-    help="Directory inside the image where the tar file containing the code will be copied to",
-)
-@click.option(
-    "-i",
-    "--image",
-    "image_config",
-    required=False,
-    multiple=True,
-    type=click.UNPROCESSED,
-    callback=ImageConfig.validate_image,
-    default=[DefaultImages.default_image()],
-    help="Image used to register and run.",
-)
-@click.option(
-    "--service-account",
-    "service_account",
-    required=False,
-    type=str,
-    default="",
-)
-@click.option(
-    "--wait-execution",
-    "wait_execution",
-    required=False,
-    is_flag=True,
-    default=False,
-    help="Whether wait for the execution to finish",
-)
-@click.option(
-    "--dump-snippet",
-    "dump_snippet",
-    required=False,
-    is_flag=True,
-    default=False,
-    help="Whether dump a code snippet instructing how to load the workflow execution using flyteremote",
-)
-@click.pass_context
-def run(
-    click_ctx,
-    file_and_workflow,
-    is_remote,
-    project,
-    domain,
-    destination_dir,
-    image_config,
-    service_account,
-    wait_execution,
-    dump_snippet,
-):
+
+class JsonParamType(click.ParamType):
+    name = "json_param"
+
+    def convert(self, value, param, ctx) -> typing.Optional[typing.Union[typing.Dict, typing.List]]:
+        if value is None:
+            return None
+        if isinstance(value, list) or isinstance(value, dict):
+            return value
+        return json.loads(value)
+
+
+class DataframeType(click.ParamType):
```

Comment:
for pandas dataframe we read the entire parquet. this is not good. It should just be transparently sent!

Reply:
I think we should just create a dummy object called
@dataclass 
class DataframeType():
   file_loc: os.PathLike

Reply:
I think this is okay, we only read it in the local case and even though the type engine will write the df to a file and read from it later, I don't think local datasets are going to large enough to really affect performance.